lua/terrortown/entities/items/item_ttt_armor.lua

```diff
@@ -2,20 +2,38 @@ if SERVER then
 	AddCSLuaFile()
 end
 
-ITEM.hud = Material("vgui/ttt/perks/hud_armor.png")
 ITEM.EquipMenuData = {
 	type = "item_passive",
 	name = "item_armor",
 	desc = "item_armor_desc"
 }
+
 ITEM.material = "vgui/ttt/icon_armor"
 ITEM.CanBuy = {ROLE_TRAITOR, ROLE_DETECTIVE}
 ITEM.oldId = EQUIP_ARMOR or 1
+ITEM.limited = false
 
-if SERVER then
-	hook.Add("ScalePlayerDamage", "TTTItemArmor", function(ply, _, dmginfo)
-		if dmginfo:IsBulletDamage() and ply:HasEquipmentItem("item_ttt_armor") then
-			dmginfo:ScaleDamage(0.7)
-		end
-	end)
+function ITEM:Equip(buyer)
+	if SERVER then
+		buyer:IncreaseArmor(GetConVar("ttt_item_armor_value"):GetInt())
+	end
 end
+
+function ITEM:Reset(buyer)
+	if SERVER then
+		buyer:DecreaseArmor(GetConVar("ttt_item_armor_value"):GetInt())
+	end
+end
+
+-- REGISTER STATUS ICONS
+if CLIENT then
+	hook.Add("Initialize", "ttt2_base_register_armor_status", function() 
```

Comment:
like explained on the discord, it does belong to the armor system. It could have been split into two PRs, but since they depend on each other, I thought this might be a valid solution.